### PR TITLE
[KBFS-1522] Keep track of stored as well as unflushed bytes

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: libkbfs/block_journal.go [hunk 1]
```diff
@@ -314,16 +358,16 @@ func (j *blockJournal) putData(
...
 	if err != nil {
 		return err
 	}
 
-	// Decremented when the put journal entry is ignored or
-	// flushed.
-	err = j.adjustUnflushedBytes(int64(len(buf)))
-	if err != nil {
-		return err
+	if putData {
```
Comment (strib) [inline libkbfs/block_journal.go]: Ah I guess this would fix a bug if we allowed dedup in the journal?

File: libkbfs/block_journal_test.go [hunk 2]
```diff
@@ -274,6 +307,38 @@ func TestBlockJournalRemoveReferences(t *testing.T) {
...
+	// block.
+	liveCounts, err := j.removeReferences(
+		ctx, kbfsblock.ContextMap{bID: {bCtx}})
+	require.NoError(t, err)
+	require.Equal(t, map[kbfsblock.ID]int{bID: 0}, liveCounts)
+	err = j.remove(bID)
+	require.NoError(t, err)
+
+	require.Equal(t, int64(0), j.getStoredBytes())
+	require.Equal(t, int64(len(data)), j.getUnflushedBytes())
```
Comment (strib) [inline libkbfs/block_journal_test.go]: This doesn't make sense to me.  According to your comments above:

> This [UnflushedBytes] should be always less than or equal to StoredBytes.

If the block is removed from storage, shouldn't the unflushed bytes be decremented too?  Or I guess this comment doesn't apply to tests where flush isn't called?

File: libkbfs/block_journal.go (hunk 1 above)
Comment (akalin-keybase) [inline libkbfs/block_journal.go]: Dedup would be a put + addReference, right? I think this fixes a different bug, if we somehow had a case where we'd retry a put for the same block.

File: libkbfs/block_journal_test.go (hunk 2 above)
Comment (akalin-keybase) [inline libkbfs/block_journal_test.go]: Yeah, this test is manually calling `remove()` (a private function) and so is violating the invariant. Normally, the flush should happen first, then the remove. Added a comment.

**Comment by akalin-keybase** [comment]
Flake is KBFS-1112.

**Pull request merged**